plugins/chat/assets/javascripts/discourse/initializers/chat-audio.js

```diff
@@ -15,6 +16,42 @@ export default {
       return;
     }
 
+    this.supportsServiceWorker = () => {
+      if (
+        !(
+          "serviceWorker" in navigator &&
+          typeof ServiceWorkerRegistration !== "undefined" &&
+          !capabilities.isAppWebview &&
+          navigator.serviceWorker.controller &&
+          navigator.serviceWorker.controller.state === "activated"
+        )
+      ) {
+        return false;
+      }
+
+      return true;
+    };
+
+    this.canPlaySound = () => {
+      return new Promise((resolve) => {
+        if (this.supportsServiceWorker()) {
+          navigator.serviceWorker.addEventListener("message", (event) => {
+            if ("canPlaySound" in event.data) {
+              resolve(event.data.canPlaySound);
+            } else {
+              resolve(false);
+            }
+          });
+
+          navigator.serviceWorker.ready.then((registration) => {
+            registration.active.postMessage({ chatSound: true });
+          });
+        } else {
+          resolve(true);
```

Comment:
In small edge cases where service workers are not supported by the browser, we are returning true which can still mean that multiple tabs would play the sound. However SW is widely supported across the major browsers.